### PR TITLE
Add image tests; only run a random subset of regression tests by defa…

### PR DESCRIPTION
**Pull request by jpwhite4**
…ult.

export REG_TEST_ALL=1 to have all tests run (this should take 15 - 20 minutes or so).

also fix the test script so that test errors cause the script to have a non-zero exit code.

### CODE REVIEW

File: open_xdmod/modules/xdmod/regression_tests/lib/Controllers/UsageChartsTest.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            $hash = $expectedHashes[$testName];
+        }
+
+        return array($testName, $reference, $hash);
+    }
+
+    public function chartSettingsProvider()
+    {
+        self::$helper = new \TestHarness\XdmodTestHelper();
+        self::$helper->authenticate('cd');
```
Comment (plessbd) [inline open_xdmod/modules/xdmod/regression_tests/lib/Controllers/UsageChartsTest.php]: do we only want to test one user or multiple?

Comment (jpwhite4) [reply open_xdmod/modules/xdmod/regression_tests/lib/Controllers/UsageChartsTest.php]: For the purposes of these tests one user is sufficient since I'm testing the different combinations of chart settings (the code paths of which are independent of the user account). The SUPReMM equivalent of these test will need to check different user accounts though.
